edgelet/iotedge/src/system.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::ffi::OsStr;
+
+use crate::error::{Error, ErrorKind};
+use aziotctl_common::{
+    get_status, get_system_logs as logs, restart, set_log_level as log_level, ServiceDefinition,
+    SERVICE_DEFINITIONS as IS_SERVICES,
+};
+
+lazy_static! {
+    static ref IOTEDGED: ServiceDefinition = {
+        let sockets: &'static [&'static str] = option_env!("IOTEDGE_HOST").map_or(&[], |_host| {
+            &["aziot-edged.mgmt.socket", "aziot-edged.workload.socket"]
+        });
+
+        ServiceDefinition {
+            service: "aziot-iotedgd.service",
+            sockets,
+        }
+    };
+    static ref SERVICE_DEFINITIONS: Vec<&'static ServiceDefinition> = {
+        let iotedged: &ServiceDefinition = &IOTEDGED;
+
+        let service_definitions: Vec<&ServiceDefinition> = std::iter::once(iotedged)
+            .chain(IS_SERVICES.iter().copied())
+            .collect();
+
+        service_definitions
+    };
+}
+
+pub struct System;
+
+impl System {
```

Comment:
is there a reason this needs to be a struct + impl block vs. just having a set of freestanding functions?

Reply:
It was cleaner to export and matched the export pattern of the other files better

Reply:
alright, fair enough.